<launch>
    <arg name="x" default="0"/>
    <arg name="y" default="0"/>
    <arg name="z" default="0"/>
    <arg name="R" default="0"/>
    <arg name="P" default="0"/>
    <arg name="Y" default="0"/>

    <arg name="gui" default="true"/>
    <arg name="debug" default="false"/>
    <arg name="verbose" default="false"/>
    <arg name="paused" default="false"/>
    <arg name="respawn_gazebo" default="false"/>

    <arg name="vehicle" default="Crp450"/>
    <arg name="world" default="$(find gazebo_crp450)/worlds/$(arg vehicle).world"/>
    <arg name="sdf" default="$(find gazebo_crp450)/models/$(arg vehicle)/$(arg vehicle).sdf"/>


    <include file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="gui" value="$(arg gui)"/>
        <arg name="world_name" value="$(arg world)"/>
        <arg name="debug" value="$(arg debug)"/>
        <arg name="verbose" value="$(arg verbose)"/>
        <arg name="paused" value="$(arg paused)"/>
        <arg name="respawn_gazebo" value="$(arg respawn_gazebo)"/>
    </include>



    <rosparam command="load" file="$(find gazebo_crp450)/config/arm_pid.yaml"/>

    <node name="dynamic_param" pkg="gazebo_crp450" type="dynamic_param"/>
    <node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure"/>
    
    <node name="rqt_multiplot" pkg="rqt_multiplot" type="rqt_multiplot">
        <param name="perspective_file" value="$(find gazebo_crp450)/config/rqt_multiplot.xml"/>
    </node>

    <!-- gazebo model -->
    <node name="$(anon vehicle_spawn)" pkg="gazebo_ros" type="spawn_model" output="screen" 
    args="-sdf -file $(arg sdf) -model $(arg vehicle) -x $(arg x) -y $(arg y) -z $(arg z) -R $(arg R) -P $(arg P) -Y $(arg Y)"/>

    <node name="gazebo_tools_init" pkg="gazebo_crp450" type="gazebo_tools_init" output="screen"/>

    <!-- <node name="arm_pid_cmd" pkg="gazebo_crp450" type="arm_pid_cmd.py"/> -->

</launch>